ecllibrary/std/DataPatterns/Profile.ecl

```diff
@@ -514,6 +527,67 @@ EXPORT Profile(inFile,
         #UNIQUENAME(_MakeAttr);
         LOCAL %_MakeAttr%(STRING attr) := REGEXREPLACE('\\.', attr, '_');
 
+        // Determine if a UTF-8 string really contains UTF-8 characters
+        #UNIQUENAME(IsUTF8);
+        LOCAL BOOLEAN %IsUTF8%(UTF8 str) := EMBED(C++)
+            if (lenStr == 0)
```

Comment:
Also worth 
#option pure
in the function

Reply:
Added!